### PR TITLE
Remove data stiching

### LINKED ISSUE #1632: Remove GraphQL data stitching functionality from atlas

### PR DESCRIPTION
**Pull request by drillprop**
Fix #1632

### CODE REVIEW

File: src/api/hooks/categoriesFeaturedVideos.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const { data, ...rest } = useGetAllCategoriesFeaturedVideosQuery({ ...opts })
+
+  return {
+    allCategoriesFeaturedVideos: data?.allCategoriesFeaturedVideos,
+    ...rest,
+  }
+}
+
+export const useCategoriesFeaturedVideos = (
+  id: string,
```
Comment (rafalpawlow) [inline src/api/hooks/categoriesFeaturedVideos.ts]: Nothing super important, but could we change arguments to be more meaningful? In this case `categoryId`

File: src/views/viewer/HomeView.tsx [hunk 2]
```diff
@@ -57,7 +56,7 @@ export const HomeView: React.FC = () => {
 
   return (
     <VideoContentTemplate cta={['popular', 'new', 'channels']}>
-      <VideoHero videoHeroData={videoHeroData} withMuteButton />
+      <VideoHero videoHeroData={videoHero || null} withMuteButton />
```
Comment (rafalpawlow) [inline src/views/viewer/HomeView.tsx]: Let's set default value in `useVideoHeroData`. WDYT?

**Review by rafalpawlow — CHANGES_REQUESTED** [review]
Except those 2 comments, looks and workd really great. I really like this change!

File: src/api/queries/videos.graphql [hunk 3]
```diff
@@ -100,35 +100,15 @@ query GetBasicVideos($where: VideoWhereInput) {
...
-    id
-    views
-  }
-}
-
 query GetMostViewedVideos($timePeriodDays: Int!, $limit: Int) {
   mostViewedVideos(timePeriodDays: $timePeriodDays, limit: $limit) {
-    id
-    views
+    ...BasicVideoFields
```
Comment (kdembler) [inline src/api/queries/videos.graphql]: We need to get the full fields here. Currently this query will not return all the details needed and then separate `GetVideo` will be executed for each of videos returned

File: src/api/hooks/video.ts [hunk 4]
```diff
@@ -121,32 +121,11 @@ export const useMostViewedVideos = (
   variables?: GetMostViewedVideosQueryVariables,
   opts?: MostViewedVideosQueryOpts
 ) => {
```
Comment (kdembler) [inline src/api/hooks/video.ts]: Cannot comment on it, but there's a `useMostViewedVideosIds` function above that doesn't make sense anymore

**Comment by kdembler** [comment]
(done!) Let's add a script to do a GraphQL introspection without the extension

**Review by rafalpawlow — APPROVED** [review]

File: src/api/hooks/channel.ts [hunk 5]
```diff
@@ -86,7 +85,17 @@ export const useChannels = (
...
-  const { data, ...rest } = useGetChannelsQuery({ ...opts, variables })
+  const { data, ...rest } = useGetChannelsQuery({
+    ...opts,
+    variables: {
+      ...variables,
+      where: {
+        ...variables?.where,
+        isCensored_eq: false,
+        isPublic_eq: true,
+      },
```
Comment (eldiegod) [inline src/api/hooks/channel.ts]: I would put `...variables?.where` after public and censored flags so we can override them easily if necessary

File: src/components/InfiniteGrids/InfiniteVideoGrid.tsx [hunk 6]
```diff
@@ -23,17 +25,20 @@ import { AdditionalLink, LoadMoreButtonWrapper } from './InfiniteGrid.styles'
 import { useInfiniteGrid } from './useInfiniteGrid'
 
 type InfiniteVideoGridProps = {
+  query?: DocumentNode
+  periodDays?: number
+  limit?: number
```
Comment (eldiegod) [inline src/components/InfiniteGrids/InfiniteVideoGrid.tsx]: I suggest adding small comments on these to clarify what they do exactly

File: src/components/InfiniteGrids/InfiniteVideoGrid.tsx [hunk 7]
```diff
@@ -23,17 +25,20 @@ import { AdditionalLink, LoadMoreButtonWrapper } from './InfiniteGrid.styles'
...
   title?: string
   titleLoader?: boolean
   videoWhereInput?: VideoWhereInput
   skipCount?: number
   ready?: boolean
   showChannel?: boolean
   className?: string
-  currentlyWatchedVideoId?: string
+  excludeId?: string
```
Comment (eldiegod) [inline src/components/InfiniteGrids/InfiniteVideoGrid.tsx:41]: also here; as this component gets more complex will serve as documentation

File: src/components/_channel/ExpandableChannelsList/ExpandableChannelsList.tsx [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return usePopularChannels({}, commonOpts)
+  } else if (queryType === 'promising') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return usePromisingChannels({}, commonOpts)
+  } else {
+    // regular channels query needs explicit limit and sorting as it's not defined by Orion
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useChannels({ limit: 15, orderBy: ChannelOrderByInput.CreatedAtAsc }, commonOpts)
+  }
```
Comment (eldiegod) [inline src/components/_channel/ExpandableChannelsList/ExpandableChannelsList.tsx:142]: Maybe a switch statement would solve the issue?

**Review by eldiegod — APPROVED** [review]
added some small comments and looks good overall

File: src/components/_channel/ExpandableChannelsList/ExpandableChannelsList.tsx (hunk 8 above)
Comment (kdembler) [reply src/components/_channel/ExpandableChannelsList/ExpandableChannelsList.tsx:142]: Switch is a conditional as well, it's the same issue

**Comment by eldiegod** [comment]
Right

**Pull request merged**